### PR TITLE
Adds performance tests

### PR DESCRIPTION
**Pull request by AmadeusW**
This PR brings the following tests:
* `CutCopyUndoPasteTest` - Cut, Copy, Paste, Undo
* `AutoformattingTest` - Autoformatting and smart indent on semicolon
* `BraceCompletionTest` - Autoformatting on closing brace
* `BasicTypingTest` 
 * Typing within a comment (doesn't trigger auto formatting)
 * Typing in a large file

Also updates .gitignore, it changed since Shabse last committed it.

### CODE REVIEW

File: PerformanceTests/DiagnosticApplication.cs [hunk 1]
```diff
@@ -17,33 +17,40 @@ internal class DiagnosticApplication : Application
...
 		private readonly Action _testMethod;
+		private readonly bool _clean;
 
 		/// <summary>
 		/// Runs benchmark and allows its inspection in the GUI
 		/// </summary>
 		/// <param name="testClass">Instace of TestBase. You may customize parameters.</param>
 		/// <param name="testMethod">Optionally, a test method to run</param>
-		internal static void Run(TestBase testClass, Action testMethod = null)
+		internal static void Run(TestBase testClass, Action testMethod = null, bool clean = false)
```
Comment (JoshVarty) [inline PerformanceTests/DiagnosticApplication.cs]: I think we can remove `clean` it appears never to be passed in as `true`.

File: PerformanceTests/Props/Snippets.cs [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+}
+";
+
+		internal static int GetCaretPositionInConsoleApp(Location completionLocation)
+		{
+			switch (completionLocation)
+			{
+				case Location.OutsideNamespace:
```
Comment (JoshVarty) [inline PerformanceTests/Props/Snippets.cs:34]: Can you add a comment stating where each location is? (eg. "Before first using", "Before the `class` keyword" etc.)

File: PerformanceTests/TestBase.cs [hunk 3]
```diff
@@ -38,18 +38,27 @@ private static void initializeRoslynForegroundThreadDataObject()
...
 		[Setup]
-		public void Setup()
+		public virtual void Setup()
 		{
 			Host = new VSEmbed.DemoApp.EditorWindow();
 			Host.Show();
+			SetupHost();
+		}
+
+		public virtual void SetupHost()
```
Comment (JoshVarty) [inline PerformanceTests/TestBase.cs:48]: Does any test **not** override `SetupHost()`? Consider making it abstract if not.

**Review by JoshVarty — CHANGES_REQUESTED** [review]

File: PerformanceTests/Props/Snippets.cs (hunk 2 above)
Comment (AmadeusW) [reply PerformanceTests/Props/Snippets.cs:34]: good idea! Done

File: PerformanceTests/TestBase.cs (hunk 3 above)
Comment (AmadeusW) [reply PerformanceTests/TestBase.cs:48]: they all do, but `SetupHost` has a body and cannot be marked abstract.

**Review by JoshVarty — APPROVED** [review]

**Pull request merged**